### PR TITLE
Misc simplifications & performance improvements

### PR DESCRIPTION
**Pull request by Bisaloo**

### CODE REVIEW

File: R/convert_to_numeric.R [hunk 1]
```diff
@@ -63,25 +63,19 @@ detect_to_numeric_columns <- function(scan_res) {
...
+    values <- values[values > 0L]
+    values <- values[names(values) != "missing"]
+    if (setequal(names(values), c("numeric", "character"))) {
+      if (values[["numeric"]] > (2 * values[["character"]])) {
         to_numeric <- c(to_numeric, col)
-      } else if (values[["numeric"]] < (2L * values[["character"]])) {
+      } else if (values[["numeric"]] < (2 * values[["character"]])) {
           warning(sprintf("In '%s' column, the number of numeric values", col),
                           " is same as the number of character values",
                   call. = FALSE)
```
Comment (Bisaloo) [inline R/convert_to_numeric.R:77]: The warning message here is inaccurate. @Karim-Mane, can you clarify what the intent was to help determine if the condition or the message should be fixed?

File: R/remove_constants.R [hunk 2]
```diff
@@ -52,7 +52,7 @@ remove_constants <- function(data, cutoff = 1L) {
   dat     <- data %>% janitor::remove_constant()
   removed <- setdiff(colnames(data), names(dat))
   if (length(removed) > 0L) {
-    add_this <- paste(removed, sep = ", ")
+    add_this <- toString(removed)
```
Comment (Bisaloo) [inline R/remove_constants.R:55]: `paste(sep =)` on a vector has no effect. Can you clarify what was intended here please? The proposed change causes a test failure so it's unclear to me what the expected behaviour was.

File: R/convert_to_numeric.R (hunk 1 above)
Comment (Karim-Mane) [reply R/convert_to_numeric.R:77]: For character columns with a certain percent of numeric values, this function will convert the column into numeric when requested by the user. Non-numeric values will then be replaced with NA.

However, there are cases where the proportion of numeric is almost similar to the proportion of character or less than 2 times, this warning is issued to alert the user. It is useful when the function is called together with other functions in a pipeable way as it is a signal for the user to take a closer look into that column.

Comment (Bisaloo) [reply R/convert_to_numeric.R:77]: Yes but the message is inaccurate.

"Less than 2 times more" is not equivalent to "the same".

File: R/remove_constants.R (hunk 2 above)
Comment (Karim-Mane) [reply R/remove_constants.R:55]: `paste` was a replacement for `glue::glue_collapse()` which uses `sep`. I mistakenly forgot to replace `sep`, with `collaspe` as I intend to convert the `removed` vector into a comma-separated list of column names. This is suitable for the display in `print_report()`.

Previously, I had the below in the corresponding test file:

```{r}
expect_identical(report[["constant_columns"]], c("event_name", "country_code", "country_name"))
```

I have corrected this to account for the comma-separated list of column names returned by `toString()`:

```{r}
expect_identical(report[["constant_columns"]], "event_name, country_code, country_name")
```

File: R/convert_to_numeric.R (hunk 1 above)
Comment (Karim-Mane) [reply R/convert_to_numeric.R]: I have updated the message in commit `3fed9d4`.

**Pull request merged**